code/game/machinery/medical_kiosk.dm

```diff
@@ -0,0 +1,345 @@
+			if(!(altPatient == W.return_patient()))
+				clearScans()
+			altPatient = W.return_patient()
+			user.visible_message("<span class='notice'>[W.return_patient()] has been set as the current patient.</span>")
+			W.selected_target = null
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		scanner_wand = O
+		return
+	return ..()
+
+/obj/machinery/medical_kiosk/AltClick(mob/living/carbon/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE))
+		return
+	if(!scanner_wand)
+		to_chat(user, "<span class='warning'>The scanner wand is currently removed from the machine.</span>")
+		return
+	if(!user.put_in_hands(scanner_wand))
+		to_chat(user, "<span class='warning'>The scanner wand falls to the floor.</span>")
+		scanner_wand = null
+		return
+	user.visible_message("<span class='notice'>[user] unhooks the [scanner_wand] from [src].</span>", \
+	"<span class='notice'>You detach the [scanner_wand] from [src].</span>")
+	playsound(src, 'sound/machines/click.ogg', 60, TRUE)
+	scanner_wand = null
+
+/obj/machinery/medical_kiosk/Destroy()
+	qdel(scanner_wand)
+	return ..()
+
+/obj/machinery/medical_kiosk/emag_act(mob/user)
+	..()
+	if(obj_flags & EMAGGED)
+		return
+	if(user)
+		user.visible_message("<span class='warning'>[user] waves a suspicious card by the [src]'s biometric scanner!</span>",
+	"<span class='notice'>You overload the sensory electronics, the diagnostic readouts start jittering across the screen..</span>")
+	obj_flags |= EMAGGED
+	var/obj/item/circuitboard/computer/cargo/board = circuit
+	board.obj_flags |= EMAGGED //Mirrors emag status onto the board as well.
+	pandemonium = TRUE
+
+/obj/machinery/medical_kiosk/examine(mob/user)
+	. = ..()
+	if(scanner_wand == null)
+		. += "<span class='notice'>\The [src] is missing its scanner.</span>"
+	else
+		. += "<span class='notice'>\The [src] has its scanner clipped to the side. Alt-Click to remove.</span>"
+
+/obj/machinery/medical_kiosk/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	var/patient_distance = 0
+	if(!ishuman(user))
+		to_chat(user, "<span class='warning'>[src] is unable to interface with non-humanoids!</span>")
+		if (ui)
+			ui.close()
+		return
+	patient_distance = get_dist(src.loc,altPatient)
+	if(altPatient == null)
+		say("Scanner reset.")
+		altPatient = user
+	else if(patient_distance>5)
+		altPatient = null
+		say("Patient out of range. Resetting biometrics.")
+		clearScans()
+		return
+
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "MedicalKiosk", name, 575, 420, master_ui, state)
+		ui.open()
+		icon_state = "kiosk_off"
+		RefreshParts()
+		H = user
+		C = H.get_idcard(TRUE)
+
+/obj/machinery/medical_kiosk/ui_data(mob/living/carbon/human/user)
+	var/list/data = list()
+	var/patient_name = altPatient.name
+	var/patient_status = "Alive."
+	var/max_health = altPatient.maxHealth
```

Comment:
I would personally prefer you put all of this in a dict. This is a lot of stats. While not strictly necessary, it'd allow you to a: have multiple patients and dodge concurrency issues (IE, what happens if two people want to scan at the same time, and they both have wands?) and b: make the code a little bit less long.

Something like:
var/list/patients_info = list()
for(patient in patients)
var/list/patient_info() = list()
patient_info["name"] = patient.name
patient_info["bleeding"] = (patient.bleeding) ? "Patient is bleeding" : "Patient is not bleeding"
patients_info[++patients_info.len] = patient_info
data["patients"] = patients_info